src/main/java/com/minecolonies/coremod/client/gui/WindowListRecipes.java

```diff
@@ -126,6 +144,13 @@ public void updateElement(final int index, @NotNull final Pane rowPane)
                 final ItemIcon icon = rowPane.findPaneOfTypeByID(OUTPUT_ICON, ItemIcon.class);
                 icon.setItem(recipe.getPrimaryOutput());
 
+                if (!isRemovalPossible) {
```

Comment:
You'd have to add this to RecipeStorage and then also add this to the serializer/deserializer this flag.

Reply:
Are RecipeStorages reused/shared using the RecipeManager? Is that maybe planned for performance reasons at one point? If so, coding readOnly into the recipe is bound to break something at some time (i.e. teaching the same recipe to two different huts). In that case, I would add builtInRecipes into the workerhuts, which would run in parallel with the current "playerRecipes". This would also align with the other suggestion about allowing/disallowing playerRecipes together with builtIn-recipes in the future. If recipes are never reused, I'll just add this flag to the recipe?

Reply:
They are reused on a global level. The question is, do we want this recipe to be ever "alterable" ?

Reply:
We're talking about the ability to remove the recipe from the building. I.e. we don't want the stripped recipes to be removed from the lumberjack. In extension, the same is true for the fletcher and the dyer btw.
But, if I happen to teach someone a recipe that another building has built-in (maybe that's something you guys are planning to use more in the future, say the sawmill can automatically and unremovably do the plank-recipe), then I should be able to remove that recipe from the building again.

Reply:
Tbf, that feature we're talking about would add it automatically to all and to new buildings. However the player would still be allowed to remove it.

(Similarly in the glassblower it is intentionally allowed to remove sand (to avoid requesting red sand to the warehouse).

Reply:
If that's the case, I'll just hide the teach-button on lumberjacks (if you even want that) and leave the rest alone?
BTW, adding to existing buildings didn't work with this method for me, but that might be because of the single player dev world ..

Reply:
Yeah, I'd be fine with hiding the remove and teach button at the lj. There is nothing we'd have to teach the lj

Reply:
So:
Option 1: Globally hide teach+remove on Lumberjacks, like it's done in the PR now.
Option 2: Show both teach+remove, like Fletcher+Dyer, but if the player removes the recipe, he/she has to rebuild the thing to recover.
Option 3: Refunction the Teach-Button to reset the built-in recipes to the initial state.

Admin's choice :)

Reply:
I'd say option one, but the naming is probably bad that's why Asher complained. Better name it:

allowRecipeAlterations()

Reply:
in the building that gets synched to the view

Reply:
Renamed in d230c91, kept teach/recipelist logic as was previously, but added logic to hide the recipe-list-button too if there's simply no recipes available for this hut (no built-in + no way to teach).